### PR TITLE
libgda6: init at 6.0.0

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
- libgda: clean up the expression
- libgda6: init at 6.0.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ 	] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
@SuperSandro2000 Could you please check if this builds on Darwin? 5.0 expression does something weird about linking sqlite.

File: pkgs/development/libraries/libgda/6.x.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+, gobject-introspection
+, vala
+, libgee
+, sqlite
+, gtk-doc
+, yelp-tools
+, mysqlSupport ? false
+, libmysqlclient ? null
+, postgresSupport ? false
+, postgresql ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgda/6.x.nix:25]:
```suggestion
, mysqlSupport ? false
, libmysqlclient
, postgresSupport ? false
, postgresql
```

that messes with overriding inputs.

File: pkgs/development/libraries/libgda/6.x.nix [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}:
+
+assert mysqlSupport -> libmysqlclient != null;
+assert postgresSupport -> postgresql != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgda/6.x.nix:29]:
```suggestion

```

File: pkgs/development/libraries/libgda/6.x.nix [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    updateScript = gnome3.updateScript {
+      packageName = pname;
+      attrPath = "libgda6";
+    };
+  };
+
+  meta = with lib; {
+    description = "Database access library";
+    homepage = "https://www.gnome-db.org/";
+    license = [ licenses.lgpl2 licenses.gpl2 ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libgda/6.x.nix:83]:
```suggestion
    license = with licenses; [ lgpl2 gpl2 ];
```

[l]gpl2Only or [l]gpl2Plus?

File: pkgs/development/libraries/libgda/6.x.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/development/libraries/libgda/6.x.nix:25]: What do you mean?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>libgda6</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 111597` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

Builds fine with the following diff on darwin

```diff
diff --git a/pkgs/desktops/gnome-3/core/yelp-xsl/default.nix b/pkgs/desktops/gnome-3/core/yelp-xsl/default.nix
index b02c7039e51..9dea1471d42 100644
--- a/pkgs/desktops/gnome-3/core/yelp-xsl/default.nix
+++ b/pkgs/desktops/gnome-3/core/yelp-xsl/default.nix
@@ -39,6 +39,6 @@ stdenv.mkDerivation rec {
     description = "Yelp's universal stylesheets for Mallard and DocBook";
     maintainers = teams.gnome.members;
     license = [licenses.gpl2 licenses.lgpl2];
-    platforms = platforms.linux;
+    platforms = platforms.unix;
   };
 }
diff --git a/pkgs/development/misc/yelp-tools/default.nix b/pkgs/development/misc/yelp-tools/default.nix
index 9d5eb2af496..4edc2aad0ba 100644
--- a/pkgs/development/misc/yelp-tools/default.nix
+++ b/pkgs/development/misc/yelp-tools/default.nix
@@ -25,6 +25,6 @@ stdenv.mkDerivation rec {
     description = "Small programs that help you create, edit, manage, and publish your Mallard or DocBook documentation";
     maintainers = with maintainers; [ domenkozar ];
     license = licenses.gpl2;
-    platforms = platforms.linux;
+    platforms = platforms.unix;
   };
 }
```

**Comment by jtojnar** [comment]
Thanks, fixed the license for `yelp-{tools,xsl}` as well.

**Pull request merged**